File: c++/learn_c++_with_robot_project/components/src/Kinematics.cpp
#include "components/Kinematics.hpp"
#include <iostream>

Kinematics::Kinematics(double x, double y, double theta_rad) {
    m_pose << x, y, theta_rad;
}

void Kinematics::move(double dx, double dy) {
    m_pose(0) += dx; 
    m_pose(1) += dy; 
}

void Kinematics::printPose() const {
    std::cout << "Robot Pose (x, y, theta): [" 
              << m_pose(0) << ", " 
              << m_pose(1) << ", " 
              << m_pose(2) << "]\n";
}

Eigen::Vector3d Kinematics::getPose() const {
    return m_pose;
}

File: python/learn_python_with_robot_project/components/kinematics.py
import numpy as np

class Kinematics():
    def __init__(self, x, y, theta_rad):
        self.__pose = np.array([x, y, theta_rad])

    def move(self, dx, dy):
        self.__pose[0] += dx
        self.__pose[1] += dy

    def get_pose(self):
        return self.__pose    
    
    def print_pose(self):
        print(f"Robot Pose (x, y, theta): [{self.__pose[0]}, {self.__pose[1]}, {self.__pose[2]}]\n")
